tests/test_packedboolarray.py

```diff
@@ -13,378 +13,1094 @@
 
 class PackedBoolArrayTestCase(unittest.TestCase):
     """Tests for _PackedBoolArray."""
-    def test_create(self):
-        m = _PackedBoolArray(size=0)
-        self.assertEqual(m.size, 0)
-
-        m = _PackedBoolArray(size=2**10)
-        self.assertEqual(m.size, 2**10)
-        self.assertEqual(len(m.data_array), 2**10 // 8)
-        testing.assert_array_equal(m.data_array, 0)
-
-        m = _PackedBoolArray(size=2**10, fill_value=True)
-        self.assertEqual(m.size, 2**10)
-        self.assertEqual(len(m.data_array), 2**10 // 8)
-        testing.assert_array_equal(m.data_array, 255)
-
-        m2 = _PackedBoolArray(data_buffer=m.data_array)
-        self.assertEqual(m2.size, 2**10)
-        self.assertEqual(len(m2.data_array), 2**10 // 8)
-        testing.assert_array_equal(m2.data_array, 255)
+    def _make_short_arrays(self):
+        # This creates 4 short (<= 8) arrays:
+        # First is the full 8 (start = 0, end = 8).
+        # Second starts at 0, ends less than 8.
+        # Third starts at >0, ends at 8.
+        # Fourth starts at >0, ends at <8.
+
+        short_arrays = []
+
+        array = np.zeros(8, dtype=np.bool_)
+        array[0] = True
+        array[4] = True
+        array[7] = True
+
+        data = np.packbits(array, bitorder="little")
+
+        short_arrays.append(_PackedBoolArray(data_buffer=data.copy(), start_index=0, stop_index=8))
+        short_arrays.append(_PackedBoolArray(data_buffer=data.copy(), start_index=0, stop_index=6))
+        short_arrays.append(_PackedBoolArray(data_buffer=data.copy(), start_index=1, stop_index=8))
+        short_arrays.append(_PackedBoolArray(data_buffer=data.copy(), start_index=1, stop_index=6))
+
+        return short_arrays
+
+    def _make_middle_arrays(self):
+        # This creates 4 middle (8 < size < 16) arrays:
+        # First is the full 16 (start = 0, end = 16).
+        # Second starts at 0, ends less than 16.
+        # Third starts at >0, ends at 16.
+        # Fourth starts at >0, ends at <16.
+
+        middle_arrays = []
+
+        array = np.zeros(16, dtype=np.bool_)
+        array[0] = True
+        array[4] = True
+        array[7] = True
+        array[12] = True
+        array[15] = True
+
+        data = np.packbits(array, bitorder="little")
+
+        middle_arrays.append(_PackedBoolArray(data_buffer=data.copy(), start_index=0, stop_index=16))
```

Comment:
The last index is 15 in this case. Another example of how the doc string and parameter names do not match how they are used.